### PR TITLE
games-engines/solarus: add 1.6.5, 9999; games-misc/solarus-quest-editor: new package

### PR DESCRIPTION
**Pull request by Kangie**

### CODE REVIEW

**Comment by Kangie** [comment]
dropped commit `games-engines/solarus: drop 1.3.1-r100`:

- games-rpg/zsdx
- games-rpg/zsxd

Current issues:

1. ~`games-engines/solarus`: tests pass consistently on `EAPI=7`, one test hangs on `EAPI=8`.~ works with `-j1`; good enough for me!
2. `games-misc/solarus-quest-editor`: segfaults on startup

The initial prompt for the `solarus-quest-editor` ebuild was a user not being able to compile it at all, so I'm calling 'immediate segfault' a win for now. :smile:

Do not merge; WIP!

**Comment by Kangie** [comment]
According to [upstream](https://www.solarus-games.org/en/games/the-legend-of-zelda-mystery-of-solarus-dx) [both](https://www.solarus-games.org/en/games/the-legend-of-zelda-mystery-of-solarus-xd) of the ebuilds that depend on 1.3.x solarus can be updated to use 1.6.x; I'll do that as part of this work so that we can drop the old version.

File: games-rpg/zsxd/zsxd-1.12.2.ebuild [hunk 1]
```diff
@@ -38,12 +39,12 @@ src_configure() {
...
-	newicon -s 48 build/icons/${PN}_icon_48.png ${PN}.png
-	newicon -s 256 build/icons/${PN}_icon_256.png ${PN}.png
+	newicon -s 48 "build/icons/${PN}_icon_48.png" "${PN}.png"
+	newicon -s 256 "build/icons/${PN}_icon_256.png" "${PN}.png"
 
 	# install proper wrapper script
-	rm "${ED}"/usr/bin/${PN} || die
-	make_wrapper ${PN} "solarus \"/usr/share/solarus/${PN}\""
+	rm "${ED}/usr/bin/${PN}" || die
+	make_wrapper ${PN} "solarus-run \"/usr/share/solarus/${PN}\""
```
Comment (thesamesam) [inline games-rpg/zsxd/zsxd-1.12.2.ebuild:47]:
```suggestion
	make_wrapper ${PN} "solarus-run \"${EPREFIX}/usr/share/solarus/${PN}\""
```

File: games-rpg/zsdx/zsdx-1.12.3.ebuild [hunk 2]
```diff
@@ -1,32 +1,29 @@
...
+SRC_URI="https://gitlab.com/solarus-games/${PN}/-/archive/v${PV}/${PN}-v${PV}.tar.gz"
+S="${WORKDIR}/${PN}-v${PV}"
 
 LICENSE="all-rights-reserved CC-BY-SA-3.0 GPL-3"
 SLOT="0"
 KEYWORDS="~amd64 ~x86"
-RESTRICT="mirror"
+RESTRICT="mirror bindist"
 
 RDEPEND="
```
Comment (thesamesam) [inline games-rpg/zsdx/zsdx-1.12.3.ebuild:19]: DEPEND too? Or no?

File: games-misc/solarus-quest-editor/files/solarus-quest-editor-1.6.5-fix-segfault.patch [hunk 3]
```diff
@@ -0,0 +1,22 @@
+Solarus quest editor will segfault on startup with -O2
+
+See: https://gitlab.com/solarus-games/solarus-quest-editor/-/commit/921ac6170a08dcfe601c7a11ee9f7c85386a4f3f
```
Comment (thesamesam) [inline games-misc/solarus-quest-editor/files/solarus-quest-editor-1.6.5-fix-segfault.patch:3]: This is fine, but if it's literally the same commit (or a trivial rebase), instead go to https://gitlab.com/solarus-games/solarus-quest-editor/-/commit/921ac6170a08dcfe601c7a11ee9f7c85386a4f3f.patch, save it as this, then scrub-patch on it (and include the link you did, with the description) so we get some more metadata.

File: games-rpg/zsdx/zsdx-1.12.3.ebuild [hunk 4]
```diff
@@ -38,12 +35,12 @@ src_configure() {
 
 src_install() {
 	cmake_src_install
-	newicon -s 48 build/icons/${PN}_icon_48.png ${PN}.png
-	newicon -s 256 build/icons/${PN}_icon_256.png ${PN}.png
+	newicon -s 48 "build/icons/${PN}_icon_48.png" "${PN}.png"
```
Comment (thesamesam) [inline games-rpg/zsdx/zsdx-1.12.3.ebuild:38]: I usually prefer to only quote things which I know may contain spaces, but this isn't wrong per se, it's just arguably "noisier". But it's a style thing.

**Comment by JinxAdnix** [comment]
Is this one going to get sent to the main branch at some point? I would really like to use it.

**Comment by thesamesam** [comment]
@Kangie Is it ready?

**Pull request closed without merging**